deprecated-claude-app/backend/src/routes/custom-models.ts

```diff
@@ -0,0 +1,318 @@
+      res.status(204).send();
+    } catch (error) {
+      console.error('Error deleting custom model:', error);
+      res.status(500).json({ error: 'Failed to delete custom model' });
+    }
+  });
+
+  // Test a custom model connection
+  router.post('/:id/test', async (req, res) => {
+    try {
+      const userId = (req as any).userId;
+      if (!userId) {
+        return res.status(401).json({ error: 'Unauthorized' });
+      }
+
+      const model = await db.getUserModel(req.params.id, userId);
+      if (!model) {
+        return res.status(404).json({ error: 'Model not found' });
+      }
+
+      console.log(`Testing custom model: ${model.displayName} (${model.provider})`);
+
+      // Import the appropriate service
+      if (model.provider === 'openrouter') {
+        const { OpenRouterService } = await import('../services/openrouter.js');
+        const { ApiKeyManager } = await import('../services/api-key-manager.js');
+        
+        const apiKeyManager = new ApiKeyManager(db);
+        const selectedKey = await apiKeyManager.getApiKeyForRequest(userId, 'openrouter', model.id);
+        
+        if (!selectedKey) {
+          return res.status(400).json({ 
+            success: false,
+            error: 'No OpenRouter API key configured. Please add an OpenRouter API key in Settings → API Keys.' 
+          });
+        }
+
+        const service = new OpenRouterService(db, selectedKey.credentials.apiKey);
+        
+        // Make a minimal test request
+        try {
+          let testResponse = '';
+          await service.streamCompletion(
+            model.providerModelId,
+            [{
+              id: 'test-msg',
+              conversationId: 'test-conv',
+              branches: [{ id: 'test-branch', content: 'Say "test successful" and nothing else.', role: 'user', createdAt: new Date() }],
+              activeBranchId: 'test-branch',
+              order: 0
+            }],
+            undefined,
+            { temperature: 0.7, maxTokens: 50 },
+            async (chunk: string, isComplete: boolean) => {
+              testResponse += chunk;
+            }
+          );
+          
+          res.json({ 
+            success: true, 
+            message: 'Connection successful!',
+            response: testResponse.slice(0, 100)
+          });
+        } catch (error: any) {
+          console.error('OpenRouter test failed:', error);
+          res.json({ 
+            success: false, 
+            error: error.message || 'Failed to connect to OpenRouter' 
+          });
+        }
+      } else if (model.provider === 'openai-compatible') {
+        const { OpenAICompatibleService } = await import('../services/openai-compatible.js');
+        
+        if (!model.customEndpoint?.baseUrl) {
+          return res.status(400).json({ 
+            success: false,
+            error: 'No endpoint configured for this model' 
+          });
+        }
+
+        const service = new OpenAICompatibleService(
```

Comment:
One issue with this:
- The server is the one that makes the connection, not the client. So it doesn't really make sense for a user to put in "localhost" (since that's localhost of the server, not the client). So localhost only works if someone is self-hosting arc chat. I don't think this is a blocker, but worth nothing as it may be confusing/may be worth adding as a future PR.